daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DPathRuntime.scala

```diff
@@ -291,7 +291,7 @@ abstract class Converter extends RecipeOp {
       try {
         computeValue(arg, dstate)
       } catch {
-        case e: NumberFormatException => {
+        case e @ (_: NumberFormatException | _: IllegalArgumentException) => {
```

Comment:
Is NumberFormatException still the right thing to catch here? In NodeInfo.scala you introduce new exception classes and I expected to see those being caught here.

Reply:
So these exceptions do not involve calls to fromXMLString/Number and still occur with the changes I have made. The computeValue calls are primarily implemented in ConverterOps.scala, such as LongToNonNegativeInteger, which are all performing very similar operations as the fromXMLString code is. Should all of these Converter classes be changed too or do we just rely on the catch here for NumberFormatException and IllegalArgumentException (which was needed for dateTime conversions)?

Reply:
I mentioend in another comment some where, but I would suggest changing the converters to all use PrimType.Foo.fromNumber(...). fromNumber will handle the validation and conversion, and will throw just your new exception that can be caught here.

Reply:
I must have missed that, I will start making those changes